web-common/src/components/button-toggle-group/ButtonToggleGroup.svelte

```diff
@@ -0,0 +1,126 @@
+<script lang="ts" context="module">
+  export type SubButtonKey = number | string;
+
+  export type ButtonGroupContext = {
+    registerSubButton?: (key: SubButtonKey) => void;
+    toggleSubButton?: (key: SubButtonKey) => void;
+    selectedKey?: Writable<SubButtonKey>;
+    firstKey?: Writable<SubButtonKey>;
+    lastKey?: Writable<SubButtonKey>;
+    disabledKeys?: SubButtonKey[];
+  };
+  export const buttonGroupContext: ButtonGroupContext = {};
+</script>
+
+<script lang="ts">
+  import { setContext, onDestroy } from "svelte";
+  import { writable, get, Writable } from "svelte/store";
+  import { createEventDispatcher } from "svelte";
+
+  // If selectionRequired is true, then a sub button must be selected at all times.
+  // In this case the button group behaves like a radio button.
+  // If selectionRequired is false, It is possible to have no sub button selected.
+  // If additionally, defaultKey is undefined, then no sub button is selected by default.
+  export let selectionRequired = false;
+
+  export let defaultKey: number | string = undefined;
+  export let disabledKeys: (number | string)[] = [];
+
+  const dispatch = createEventDispatcher();
+
+  const subButtons = [];
+  const selectedKey = writable(null);
+
+  // if a default key is provided, then select it if it is not disabled
+  // this Must be run reactively in case the default key changes or
+  //  the set of disabled keys changes
+  $: if (defaultKey !== undefined && !disabledKeys.includes(defaultKey)) {
+    selectedKey.update(() => defaultKey);
+  } else {
+    selectedKey.update(() => null);
+  }
+
+  const firstKey = writable(null);
+  const lastKey = writable(null);
```

Comment:
Can't these just be derived from subButtons? like firstKey = subButtons.at(0), lastKey = subButtons.at(-1)

Reply:
Yep you're right. Made that change.